app/controllers/likes_controller.rb

```diff
@@ -1,18 +1,27 @@
 class LikesController < ApplicationController
-  before_action :authenticate_user!
   before_action :set_post
+  before_action :log_in_user?
 
   def create
     current_user.likes.create(post_id: @post.id)
   end
 
   def destroy
-    current_user.likes.find_by(post_id: @post.id).delete
+    current_user.likes.find_by(post_id: @post.id,id: params[:id]).delete
```

Comment:
ここ記述が冗長になってます！
下記のいずれかで行けるかと思います。

```suggestion
    current_user.likes.find_by(post_id: @post.id)
```
こっちは、current_user.likesで、likeのuser_idは指定できているので、post_idで一つに絞り込んであげてるイメージ

```suggestion
    like = Like.find_by(id: params[:id])
    like.destroy
```
こっちは、単にlikeの主キーで絞り込んでるイメージ